### ISSUE TITLE
podman, signals, 

### ISSUE THREAD

**Issue by edsantiago** [projectatomic/libpod#426]
This isn't a bug report; it's a pathetic plea for help. A desperate cry into the ether, hoping it will trigger an Aha! or a Hmmm from someone.

TL;DR podman containers sometimes aren't acknowledging or propagating signals.

Run this in one shell:

    # podman run --attach=stdout --sig-proxy=true --tty=false  registry.access.redhat.com/rhel7/rhel:latest bash -c 'for NUM in `seq 1 64`; do trap "echo Received $NUM, ignoring..." $NUM; done; echo READY; while :; do sleep 0.1; done'
    READY

Then in another:

    # CID=$(podman ps |tail -1|cut -c1-4)                  ! I know, I know
    # PID=$(podman inspect --format '{{.State.Pid}}' $CID)
    # kill -22 $PID

This produces the expected output back in terminal one:

    Received 22, ignoring...

Same with 3, 2, 23, 24, 10. This is exactly as expected. It's good.

When the same `podman run` command is run within docker-autotest (super complicated pythony python framework), autotest never sees the `Received XX, ignoring` output. It does see `READY`. 

According to strace, podman under docker-autotest is receiving the signal:

    epoll_wait(4,

then, on signal:

    epoll_wait(4, 0x7fff1d46a060, 128, -1)  = -1 EINTR (Interrupted system call)
    --- SIGUSR1 {si_signo=SIGUSR1, si_code=SI_USER, si_pid=1302, si_uid=0} ---
    rt_sigreturn({mask=[]})                 = -1 EINTR (Interrupted system call)
    epoll_wait(4,

I've also tried `grep Sig /proc/$PID/status` and see nothing suspicious.

It doesn't seem to be tty-related: I've tried (command-line) `</dev/null` or `|cat` or `>/tmp/mylog`. Same successful behavior. It doesn't seem to be a buffering issue: `podman logs $CID` shows the Received lines when run from terminal, and does not when run from docker-autotest (i.e. same behavior under docker-autotest).

The final bit of weirdness: when running under docker-autotest, the first `podman inspect` returns the pid of the `podman` command. This seems normal and expected. But when I'm poking around in another term, while docker-autotest is running and waiting (in vain) for output, the same `podman inspect` returns a different PID - the bash command started above, but now a child of `conmon`.
```
root      8722  0.0  0.0  86080  2208 ?        Ssl  22:41   0:00 /usr/libexec/crio/conmon -c 867dbb8f2b6f5f0a880f729f3247f40b7b2300fa102f0e2ae242cca567945695 -u 867dbb8f2b6f5f0a880f729f3247f40b7b2300fa102f0e2ae242cca567945695 -r /usr/bin/runc -b /var/lib/containers/storage/overlay-containers/867dbb8f2b6f5f0a880f729f3247f40b7b2300fa102f0e2ae242cca567945695/userdata -p /var/run/containers/storage/overlay-containers/867dbb8f2b6f5f0a880f729f3247f40b7b2300fa102f0e2ae242cca567945695/userdata/pidfile -l /var/lib/containers/storage/overlay-containers/867dbb8f2b6f5f0a880f729f3247f40b7b2300fa102f0e2ae242cca567945695/userdata/ctr.log --exit-dir /var/run/libpod/exits --socket-dir-path /var/run/libpod/socket -t
root      8738  0.2  0.0  11784  2660 pts/0    Ss+  22:41   0:00  \_ bash -c for NUM in `seq 1 64`; do trap "echo Received $NUM, ignoring..." $NUM; done; echo READY; while :; do sleep 0.1; done
root     12156  0.0  0.0   4328   628 pts/0    S+   22:46   0:00      \_ sleep 0.1
```
Sending a signal to that process (`kill -10 8738`) causes docker-autotest to see the expected `Received` line and move on to the next test.

I'm sorry for the poor quality of this report. I'm just really lost.

**Comment by mheon** [comment]
We don't have `--sig-proxy` implemented yet. It should probably pop up a warning or something to that effect to prevent confusion.

Or finish implementing signal handling... I might take a crack at it if I can find the time.

So this is fully expected, because our signal handling doesn't work, podman itself eats the signals and the container doesn't get them

**Comment by mheon** [comment]
The inspect issue is strange, though. I would think we should report the container PID always? It also should never change once first set, I think, though I'm less sure of that

**Comment by mheon** [comment]
I also suspect --attach {stdin, stdout, stderr} are all nonfunctional. We presently don't support nonterminal attach. Another thing that needs docs.
